### PR TITLE
fix: write lock for warmup

### PR DESCRIPTION
**Pull request by xgui3783**
Since warmup writes files, it is potentially possible to have multiple process trying to warmup simultaneously. 

This PR prevents simultaneous write by using a FileLock (or SoftFileLock in windows).

### CODE REVIEW

File: siibra/retrieval/cache.py [hunk 1]
```diff
@@ -23,6 +23,12 @@
 from typing import Callable, List, NamedTuple, Union
 from concurrent.futures import ThreadPoolExecutor
 from pathlib import Path
+import platform
+
+if platform.system() == "Linux":
+    from filelock import FileLock as Lock
+else:
+    from filelock import SoftFileLock as Lock
```
Comment (AhmetNSimsek) [inline siibra/retrieval/cache.py:31]: I think we can just use FileLock regardless: https://py-filelock.readthedocs.io/en/latest/index.html#filelock-vs-softfilelock

**Review by AhmetNSimsek — APPROVED** [review]
Just one comment, otherwise LGTM

File: siibra/retrieval/cache.py (hunk 1 above)
Comment (xgui3783) [reply siibra/retrieval/cache.py:31]: copied directly from https://github.com/FZJ-INM1-BDA/siibra-python/blob/main/siibra/retrieval/requests.py#L47-L50

I thought there was an issue with FileLock on windows?

Comment (AhmetNSimsek) [reply siibra/retrieval/cache.py:31]: I can't find what could be the reason. It seems that is also your code; do you recall anything about it?

Comment (xgui3783) [reply siibra/retrieval/cache.py:31]: yea, me neither. 

Do you want me fix both instances?

Comment (AhmetNSimsek) [reply siibra/retrieval/cache.py:31]: Perhaps previously you saw the old api doc? https://github.com/tox-dev/filelock/issues/132

**Pull request merged**